### PR TITLE
build: parallel build

### PR DESCRIPTION
**Pull request by sxzz**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: build/full-bundle.ts [hunk 1]
```diff
@@ -33,65 +34,56 @@ import { getExternals } from './utils'
...
   })
 
-  const rewriter = (id) => {
+  const rewriter = (id: string) => {
     if (id.startsWith(`${EP_PREFIX}/components`))
       return id.replace(`${EP_PREFIX}/components`, './components')
-    if (id.startsWith(EP_PREFIX) && excludes.every((e) => !id.endsWith(e)))
+    else if (id.startsWith(EP_PREFIX) && excludes.every((e) => !id.endsWith(e)))
       return id.replace(EP_PREFIX, '.')
+    else return id
```
Comment (JeremyWuuuuu) [inline build/full-bundle.ts:83]: With `return` here there is actually no need to use `else`. Even though it is semantically correct.

**Comment by JeremyWuuuuu** [comment]
TBH, when there is no test cases against the bundled product, I think we better not get our hands on the bundling script, since the previous version is tested by releasing multiple versions. If you do need to add some trivial change like abstracting the `chalk` methods, I am fine with it, but if you want to change those script for bundling, I suggest you not to. I am not saying that we are afraid of making mistakes, I am saying that at the stage of a project used by thousands we should be careful handling stuff like this, if we can guarantee the quality of the product I would be fine with it(since we can't).

File: build/full-bundle.ts (hunk 1 above)
Comment (sxzz) [reply build/full-bundle.ts:83]: If don’t return, typescript will add `undefined` to the return value.

File: build/build-helper.ts [hunk 2]
```diff
@@ -1,8 +1,8 @@
-/* eslint-disable @typescript-eslint/no-var-requires */
 import helper from 'components-helper'
 import path from 'path'
 import { epRoot } from './paths'
 
+// eslint-disable-next-line @typescript-eslint/no-var-requires
 const { name, version } = require(path.resolve(epRoot, './package.json'))
 
 import icon from '../website/icon.json'
```
Comment (JeremyWuuuuu) [inline build/build-helper.ts:8]: I think you need to rebase on the `dev` for getting the newest code.

**Comment by sxzz** [comment]
@JeremyWuuuuu PTAL.

File: build/helper.ts [hunk 3]
```diff
@@ -1,49 +1,24 @@
...
+const reComponentName: InstallOptions['reComponentName'] = (title: string) =>
+  `el-${title
     .replace(/\B([A-Z])/g, '-$1')
     .replace(/[ ]+/g, '-')
     .toLowerCase()}`
-}
 
-function reDocUrl(fileName, header) {
+const reDocUrl: InstallOptions['reDocUrl'] = (fileName, header) => {
   const docs = 'https://element-plus.org/#/en-US/component/'
```
Comment (JeremyWuuuuu) [inline build/helper.ts:14]: Actually here we need to update to
```ts
const docs = 'https://element-plus.org/en-US/component/'
```

But we can do that in separate PRs.

File: build/utils.ts [hunk 4]
```diff
@@ -1,54 +1,16 @@
...
+//   const workspacePkgs = (await getWorkspaceNames()).filter((pkg) =>
+//     pkg.startsWith(EP_PREFIX)
+//   )
+//   return (module: Module) => (id: string) => {
+//     const config = buildConfig[module]
+//     if (workspacePkgs.some((pkg) => id.startsWith(pkg)))
+//       return id.replace(EP_PREFIX, config.bundle.path)
+//     else return ''
+//   }
+// }
```
Comment (JeremyWuuuuu) [inline build/utils.ts:16]: Why are we keeping this file anyway?

**Review by jw-foss — APPROVED** [review]
Approving as-is

**Comment by JeremyWuuuuu** [comment]
@sxzz got some conflicts, you can merge this PR after resolving them.
